### PR TITLE
[FLINK-4923] [Metrics] Initializes patch for Expose input/output queue length and bufferPool usage as a metric

### PR DESCRIPTION
**Pull request by zhuhaifengleon**
Initializes all task's gauge metrics in one method of TaskIOMetricGroup for consolidating the metrics in one place. Extract the used buffers method for BufferPool for usage metrics.

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java [hunk 1]
```diff
@@ -111,6 +111,11 @@ public int getNumBuffers() {
 	}
 
 	@Override
+	public int getNumberOfUsedBuffers() {
+		return numberOfRequestedMemorySegments - getNumberOfAvailableMemorySegments();
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java]: instead of `getNumberOfAvailableMemorySegments()` we could use `availableMemorySegments.size()` so we never enter a synchronized block and affect the LocalBufferPool.

File: flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/TaskIOMetricGroup.java [hunk 2]
```diff
@@ -80,18 +76,25 @@ public Meter getNumBytesOutRateMeter() {
...
+		buffers.gauge("outputQueueLength", new OutputBuffersGauge(task));
+		buffers.gauge("inPoolUsage", new InputBufferPoolUsageGauge(task));
+		buffers.gauge("outPoolUsage", new OutputBufferPoolUsageGauge(task));
+	}
+
+	/**
 	 * Input received buffers gauge of a task
 	 */
-	public static final class InputBuffersGauge implements Gauge<Integer> {
+	private final class InputBuffersGauge implements Gauge<Integer> {
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/TaskIOMetricGroup.java]: why did you remove the `static` keyword?

Comment (zhuhaifengleon) [inline flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/TaskIOMetricGroup.java]: InputBuffersGauge is unaccessible for any class except TaskIOMetricGroup, so I change the public to private. It can work no matter whether it is static or not. to save outclass reference, static is prefer.

**Comment by zentol** [comment]
will merge this later on
